Do not add -latomic ever; macOS does not have it, not even on ARM.
Reverts https://github.com/swoole/swoole-src/commit/741ace2990c5f22f098af9f874bebbdf74a342a5
--- config.m4.orig	2022-04-17 05:45:24.000000000 -0500
+++ config.m4	2022-04-18 15:34:12.000000000 -0500
@@ -784,10 +784,6 @@
         CXXFLAGS="$CXXFLAGS -std=c++11"
     fi
 
-    if test "$SW_CPU" = "arm"; then
-        PHP_ADD_LIBRARY(atomic, 1, SWOOLE_SHARED_LIBADD)
-    fi
-
     PHP_ADD_BUILD_DIR($ext_builddir/ext-src)
     PHP_ADD_BUILD_DIR($ext_builddir/src/core)
     PHP_ADD_BUILD_DIR($ext_builddir/src/memory)
